ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java

```diff
@@ -164,9 +166,14 @@ public void dispatchMessage(final EthPeer peer, final EthMessage ethMessage) {
     dispatchMessage(peer, ethMessage, protocolName);
   }
 
-  private void reattemptPendingPeerRequests() {
+  @VisibleForTesting
+  void reattemptPendingPeerRequests() {
     synchronized (this) {
-      pendingRequests.removeIf(PendingPeerRequest::attemptExecution);
+      try {
+        pendingRequests.removeIf(PendingPeerRequest::attemptExecution);
+      } catch (ConcurrentModificationException ex) {
+        LOG.warn("Failed execution of pending requests: {}", ex.getClass().getName());
+      }
```

Comment:
Is it correct to says that if the ConcurrentModificationException occurs, it stops processing the following pending requests in the list?
In that case could make sense to catched the exception in `PendingPeerRequest.attemptExecution` and return false there, so the processing of following item will be done.

Reply:
Hey Fabio, not really, CopyOnWriteArrayList will test the filter for each item first and mark those who need to be removed and then throw the exception if the list was modified by any filter. That means that it will try to execute all the requests first and then remove then from the list.  

It is not the request itself that causes the exception, but the fact that it modifies the list during the removeIf. The following snippet replicates it: https://onecompiler.com/java/3ybxwtufc

I will try to modify this logic so we first try the requests and then remove them. It should avoid the expection

Reply:
Replacing removeIf with an iterator seems to not throw the exception

Reply:
looks good with the iterator!